AuthenticatorShared/Core/Vault/Repositories/AuthenticatorItemRepository.swift

```diff
@@ -89,35 +90,112 @@ protocol AuthenticatorItemRepository: AnyObject {
-    ///   - cryptographyService
+    ///   - authenticatorItemService: Service to from which to fetch locally stored Authenticator items.
+    ///   - configService: Service to determine if the sync feature flag is turned on.
+    ///   - cryptographyService: Service to encrypt/decrypt locally stored Authenticator items.
+    ///   - sharedItemService: Service to fetch items from the shared CoreData store - shared from
+    ///     the main Bitwarden PM app.
+    ///   - errorReporter: Error Reporter for any errors encountered
+    ///   - timeProvider: A protocol wrapping the present time.
+    ///   - totpService: A service for refreshing TOTP codes.
     init(
         authenticatorItemService: AuthenticatorItemService,
+        configService: ConfigService,
         cryptographyService: CryptographyService,
         errorReporter: ErrorReporter,
+        sharedItemService: AuthenticatorBridgeItemService,
         timeProvider: TimeProvider,
         totpService: TOTPService
     ) {
         self.authenticatorItemService = authenticatorItemService
+        self.configService = configService
         self.cryptographyService = cryptographyService
         self.errorReporter = errorReporter
         self.timeProvider = timeProvider
         self.totpService = totpService
+        self.sharedItemService = sharedItemService
     }
 
     // MARK: Private Methods
 
+    /// Combine sections that are locally stored with the list of the sections created with the shared items,
+    /// when sync with the PM app is enabled.
+    ///
+    /// Note: If the `enablePasswordManagerSync` feature flag is not enabled, or if the user has not yet
+    /// turned on sync for any accounts, this method simply returns `localSections`.
+    ///
+    /// - Parameters:
+    ///   - localSections: The [ItemListSection] sections for the items locally stored
+    ///   - sharedItems: The shared items that are coming in via sync with the PM app
+    /// - Returns: A list of the sections to display in the item list
+    ///
+    private func combinedSections(
+        localSections: [ItemListSection],
+        sharedItems: [AuthenticatorBridgeItemDataView]
+    ) async throws -> [ItemListSection] {
+        guard await configService.getFeatureFlag(.enablePasswordManagerSync),
+              await sharedItemService.isSyncOn() else {
+            return localSections
+        }
+
+        let groupedByAccount = Dictionary(
+            grouping: sharedItems,
+            by: { item in
+                if let email = item.accountEmail, !email.isEmpty {
+                    if let domain = item.accountDomain, !domain.isEmpty {
+                        return email + " | " + domain
+                    } else {
+                        return email
+                    }
+                } else if let domain = item.accountDomain, !domain.isEmpty {
+                    return domain
+                } else {
+                    return ""
+                }
```

Comment:
🎨  You can achieve the same by just using `compactMap` with `joined` 😄 :
```suggestion
                return [item.accountEmail, item.accountDomain]
                    .compactMap({ $0?.nilIfEmpty })
                    .joined(separator: " | ")
```

Reply:
👍 Oh that's nice! 😄